### ISSUE TITLE
ToggleGroupControlOption: aria-label cannot be overwritten

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#35381]
### Description

It would be better if we could override the aria-label when the visual label is non-ideal for screen readers:

https://github.com/WordPress/gutenberg/blob/d4794d6364c496f82c937e7816c24bb89c8d6da1/packages/components/src/toggle-group-control/toggle-group-control-button.tsx#L45-L48

The aria-label currently defaults to the visual label, which is possibly also redundant because the same label appears as a child element.

### Step-by-step reproduction instructions

For example:

```jsx
<ToggleGroupControlOption
	key={ value }
  label="A→Z"
  aria-label="Ascending"
  value={ value }
/>
```

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
